src/lib/wallet/GoodWalletProvider.js

```diff
@@ -264,14 +271,19 @@ export const GoodWalletProvider = ({ children, disableLoginAndWatch = false }) =
 
         log.debug('switchNetwork: starting watchBalanceAndTXs', { lastBlock, network, contractsNetwork })
 
-        goodWallet.watchEvents(parseInt(lastBlock), toBlock =>
-          userStorage.userProperties.set('lastBlock_' + goodWallet.networkId, parseInt(toBlock)),
-        )
+        goodWallet
+          .watchEvents(parseInt(lastBlock), toBlock =>
+            userStorage.userProperties.set('lastBlock_' + goodWallet.networkId, parseInt(toBlock)),
+          )
+          .catch(e => {
+            log.error('watchEvents failed:', e.message, e, { contractsNetwork, network })
```

Comment:
what is that? watchevents should not fail on any network. it should just not try to watch contracts/events that are not available

Reply:
done